src/test/java/hudson/plugins/git/GitPublisherTest.java

```diff
@@ -682,6 +686,30 @@ public void testMergeAndPushWithSystemEnvVar() throws Exception {
         checkEnvVar(project, envName, envValue);
     }
 
+    @Test
+    public void testTimeout() throws Exception {
+        FreeStyleProject project = setupSimpleProject("master");
+        GitSCM scm = new GitSCM(
+            remoteConfigs(),
+            Collections.singletonList(new BranchSpec("*")),
+            false, Collections.<SubmoduleConfig>emptyList(),
+            null, null,
+            Collections.<GitSCMExtension>emptyList());
+        scm.getExtensions().add(new PreBuildMerge(new UserMergeOptions("origin", "integration", null, MergeCommand.GitPluginFastForwardMode.FF, 1234)));
+        scm.getExtensions().add(new LocalBranch("integration"));
+        project.setScm(scm);
```

Comment:
@MarkEWaite the scm extension successfully shows the timeout as "1234". However in the commit log it still displays the default.